src/include/reconcile_inline.h

```diff
@@ -378,24 +378,41 @@ __wt_rec_cell_build_addr(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_ADDR *add
     }
 
     /*
-     * We don't copy the data into the buffer, it's not necessary; just re-point the buffer's
-     * data/length fields.
+     * Encode the page stats into the address cookie of the cell if the feature flag is set. In this
+     * scenario, the address cookie will have the following layout: block manager address cookie
+     * length, block manager address cookie, page stat address cookie length, page stat address
+     * cookie.
+     *
+     * Don't copy the data into the buffer in simple cases where there are no page stats information
+     * to be encoded, it's not necessary; just re-point the buffer's data/length fields.
      */
     if (vpack == NULL) {
         WT_ASSERT(session, addr != NULL);
-        val->buf.data = addr->addr;
-        val->buf.size = addr->size;
-        val->cell_len = __wt_cell_pack_addr(
-          session, &val->cell, cell_type, recno, page_del, &addr->ps, ta, val->buf.size);
+
+        if (__wt_process.page_stats_2022 && WT_PAGE_STAT_VALID(&addr->ps))
```

Comment:
Here the same

Reply:
Fixed here: [7fdd194](https://github.com/wiredtiger/wiredtiger/pull/8792/commits/7fdd194a5742efa7716451b29b381812ed336f2c#diff-5b8d4c04a02cd50ac6b7caf1af10ca16a9a4ea7aeae4e64d5e8bb1b6fa67e645)